openmc/settings.py

```diff
@@ -795,49 +870,46 @@ def survival_biasing(self, survival_biasing):
 
         self._survival_biasing = survival_biasing
 
-
     @weight.setter
-    def weight(self, weight, weight_avg):
-
+    def weight(self, weight):
         if not is_float(weight):
             msg = 'Unable to set weight cutoff to non-floating point ' \
                   'value {0}'.format(weight)
             raise ValueError(msg)
 
-        elif not is_float(weight_avg):
-            msg = 'Unable to set weight avg. to non-floating point ' \
-                  'value {0}'.format(weight_avg)
-            raise ValueError(msg)
-
         elif weight < 0.0:
             msg = 'Unable to set weight cutoff to negative ' \
                   'value {0}'.format(weight)
             raise ValueError(msg)
 
+        self._weight = weight
+
+    @weight_avg.setter
+    def weight_avg(self, weight_avg):
+        if not is_float(weight_avg):
+            msg = 'Unable to set weight avg. to non-floating point ' \
+                  'value {0}'.format(weight_avg)
+            raise ValueError(msg)
         elif weight_avg < 0.0:
             msg = 'Unable to set weight avg. to negative ' \
                   'value {0}'.format(weight_avg)
             raise ValueError(msg)
 
-        self._weight = weight
         self._weight_avg = weight_avg
 
-
     @entropy_dimension.setter
     def entropy_dimension(self, dimension):
-
         if not isinstance(dimension, (tuple, list)):
             msg = 'Unable to set entropy mesh dimension to {0} which is ' \
                   'not a Python tuple or list'.format(dimension)
             raise ValueError(msg)
 
-        elif len(dimension) < 3 or len(dimension) > 3:
+        elif len(dimension) == 3:
```

Comment:
ditto ``!=`` rather than ``==`` - this breaks "examples/python/pincell/build-xml.py".